extension/lib/shopgate/Concurrency.js

```diff
@@ -0,0 +1,48 @@
+const { sleep } = require('./concurrency/util')
+const { ShopgateConcurrencyLock, useStorage, validLockExists } = require('./concurrency/Lock')
+
+class ShopgateConcurrency {
```

Comment:
this class has no state at all. all its methods could be static. but then again, why do we need a class at all? could as well just export a bunch of functions...

Reply:
to me it looked nice from the perspective of a caller/consumer of the code, and I do like to have an entry point to the feature/domain as it helps in dealing with dependencies

Reply:
hmm, well i'm not fully convinced, but ok